frontend/test/metabase/scenarios/collections/bookmarks.cy.spec.js

```diff
@@ -0,0 +1,92 @@
+import _ from "underscore";
+import { restore, sidebar } from "__support__/e2e/cypress";
+
+describe("Bookmarks in a collection page", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("cannot add bookmark to root collection", () => {
+    cy.visit("/collection/root");
+
+    cy.icon("bookmark").should("not.exist");
+  });
+
+  it("can add and remove bookmarks from collection from its page", () => {
+    cy.visit("/collection/1");
+
+    // Add bookmark
+    cy.icon("bookmark").click();
+
+    // Bookmark Icon should be blue once Collection is bookmarked
+    cy.icon("bookmark")
+      .should("have.css", "color")
+      .and("eq", "rgb(80, 158, 227)");
+
+    sidebar().within(() => {
+      cy.findByText("Bookmarks");
```

Comment:
We can (and should) rely more on a11y. Luckily, both "Bookmarks" and "Collections" are `H4`. I'd definitely utilize that for this test. It might seem fragile, but in this case it's good that it is. Because if we ever decide to NOT use HTML heading tags for this, test **should** fail and alert us, imo.
```js
function getSectionTitle(title) {
  return cy.findAllByRole("heading", { name: title });
}

// use it in test
sidebar().within(() => {
  getSectionTitle("Bookmarks");
  cy.findByText("Bobby Tables's Personal Collection");

  // Once there is a list of bookmarks,
  // we add a heading to the list of collections below the list of bookmarks
  getSectionTitle("Collections");
});
```

Reply:
👍, updated